crates/vqueues/src/scheduler/drr.rs

```diff
@@ -377,17 +377,22 @@ where
                 if qstate.remove_from_unconfirmed_assignments(item_hash) {
                     // drop the already acquired permit
                     let _ = self.unconfirmed_capacity_permits.split(1);
-                } else if qstate.notify_removed(item_hash) {
+                }
+
+                qstate.notify_removed(item_hash);
+
+                // if we aren't eligible check whether the removed item has made us eligible
+                // again (e.g. by releasing a concurrency token or by removing the head of the
+                // queue).
+                if !self.eligible.contains(&qid) {
```

Comment:
This seems to overlap with the change in https://github.com/restatedev/restate/pull/4066/files#diff-02cc9fd99c15c55730f0822a024e9d2c6bd7f4f92d3cc2d8c9d030773f1d6f6d